src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+            throw new IllegalArgumentException("null position");
+        }
+        setAlignmentPosition(position.getContig(), position.getStart());
+    }
+
+    /**
+     * Gets the read length.
+     *
+     * <p>Note: this is not necessarily the same as the number of reference bases the read is aligned to.
+     *
+     * @return The number of bases in the read
+     */
+    int getLength();
+
+    /**
+     * Checks if the read has bases.
+     *
+     * <p>Default implementation returns {@code getLength() == 0}.
+     *
+     * @return {@code true} if the read has no bases; {@code false} otherwise.
+     */
+    default boolean isEmpty() {
+        return getLength() == 0;
+    }
+
+    /**
+     * Gets the PHERD scaled mapping quality.
+     *
+     * <p>The {@link ReadConstants#UNKNOWN_MAPPING_QUALITY} implies valid mapping, but hard to compute quality.
+     *
+     * <p>Equivalent to <b>MAPQ</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @return mapping quality.
+     */
+    int getMappingQuality();
+
+    /**
+     * Sets the PHRED scaled mapping quality.
+     *
+     * <p>Equivalent to <b>MAPQ</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @param mapq mapping quality.
+     */
+    void setMappingQuality(final int mapq);
+
+    /**
+     * Gets the {@link Cigar} object describing how the read aligns to the reference.
+     *
+     * <p>This method should make a defensive copy of the {@link Cigar} before returning it to allow
+     * modification of the returned {@link Cigar} without effects on the {@link Read}.
+     *
+     * @return Cigar object for the read; empty cigar if there is none.
+     */
+    Cigar getCigar();
+
+    /**
+     * Gets the number of cigar elements (number + operator) in the cigar string.
+     *
+     * <p>Default implementation returns {@code getCigar().numCigarElements()}.
+     * Subclasses may override to provide more efficient implementations.
+     *
+     * @return number of cigar elements (number + operator) in the cigar string.
+     */
+    default int getCigarLength() {
+        return getCigar().numCigarElements();
+    }
+
+    /**
+     * Sets the {@link Cigar} object describing how the read aligns to the reference.
+     *
+     * @param cigar Cigar object for the read; empty Cigar if there is none.
+     *
+     * @throws IllegalArgumentException if the cigar is {@code null}.
+     */
+    void setCigar(final Cigar cigar);
+
+    /**
+     * Gets the contig name where the read's mate is mapped to. May return null if there is no unique mapping.
+     *
+     * <p>Equivalent to <b>RNEXT</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @return name of the contig the read's mate is mapped to; {@link ReadConstants#NO_ALIGNMENT_REFERENCE_NAME} if unset (e.g. for unmapped reads).
+     */
+    String getMateContig();
```

Comment:
Using `next` to allow multiple segments would be nice. Let's see what the rest of the community things. Maybe we should point an example for pair-end reads too, because is the most common next-template, I guess...

Reply:
I realized that this will be important also for the flag methods (isPaired, isMateUnmapped). Maybe we can have also sugar-syntax for pair-end apart of multiple segments...